### ISSUE TITLE
Ansible Homebrew Modules 

### ISSUE THREAD

**Issue by Shashikant86** [geerlingguy/ansible-role-homebrew#62]
Just wondering whats difference between this role and Ansible Homebrew module? 
http://docs.ansible.com/ansible/homebrew_module.html

Are we doing something different here ?

**Comment by elliotweiser** [comment]
@Shashikant86 I suggest you take a look at https://github.com/geerlingguy/ansible-role-homebrew/blob/master/tasks/main.yml.

This role actually uses the Ansible `homebrew` module to manage homebrew packages. However, this role does a lot more:

1. Ensures that homebrew is installed and in the desired location
2. Manages homebrew packages via the homebrew module
3. Manages homebrew casks via the homebrew_cask module
4. Configures taps
5. Upgrades packages
6. Install packages from a Brewfile

In short, it's a collection of homebrew-related tasks to help simplify homebrew management. It's a _superset_ of the functionality of the Ansible `homebrew` module.

**Comment by geerlingguy** [comment]
@Shashikant86 - Exactly what @elliotweiser said! It's also the basis for my own Mac management (three Macs, using the same configuration, for perfect harmony and easy rebuild) using https://github.com/geerlingguy/mac-dev-playbook

**Comment by Shashikant86** [comment]
@geerlingguy @elliotweiser Thanks a lot for your answers. It's amazing !! Trying to build similar kinda env for iOS development and taken inspiration from @geerlingguy Mac-Dev-Playbook an written my own role on top of that :) 
https://github.com/Shashikant86/ansible-ios-ci

Cheers guys.. Keep up amazing work.  👍

**Comment by Shashikant86** [comment]
@geerlingguy I couldn't manage to find Ansible way to install Xcode but I have provided workaround in this role .. Le the know if you have better solutions cc// @elliotweiser :)

**Comment by elliotweiser** [comment]
@Shashikant86 Check out https://github.com/mas-cli/mas. I believe @geerlingguy wrote a [role][role] for installing/utilizing that tool. I've poked into `mas` a little bit and it has some shortcomings, but I think you'll be able to get out of it what you want with little elbow grease 😄.

[role]: https://github.com/geerlingguy/ansible-role-mas
